### PR TITLE
Allow setting a custom Random

### LINKED ISSUE #91: Allow setting a seed for Random
I'm using [Kotest property based testing](https://kotest.io/docs/proptest/property-based-testing.html), and this needs a deterministic way of generating random data based on a seed, or existing `Random` instance.

iban4j can generate random IBANs, but I can't set a seed or pass in a `Random` instance.

I would like to add an overload for the `Iban.random()` method that will accept a `Random` parameter. 

I can try making a PR for this, if you agree. Thanks!

### PR DESCRIPTION
**Pull request by aSemy**
Allow passing in a custom random, so users can make the generation deterministic.

resolves #91 

#### Summary

- fix spelling mistake in test, `ibanContruction` -> `ibanConstruction`
- add `toString()` to BbanStructureEntry.java (this helps the test error messages)
- tests to check if randomness is deterministic
- add additional tests for validity of `BbanStructureEntry.java`

#### TODO

- [x] I was confused by the `EntryCharacterType` - need to remove the TODO comment and possibly update the Javadoc

### CODE REVIEW

File: src/main/java/org/iban4j/bban/BbanStructureEntry.java [hunk 1]
```diff
@@ -104,12 +103,27 @@ public int getLength() {
...
+        /**
+         * Alphabetical characters (A-Z only)
+         */
+        a,
+        // TODO 'c' used to say:  // upper and lower case alphanumeric characters (A-Z, a-z and 0-9)
+        //      but only generates A-Z0-9. Which is correct? Should it generate a-z too?
+        /**
+         * Combined alphabetical (uppercase) and numeric characters (A-Z and 0-9)
+         */
+        c
```
Comment (aSemy) [inline src/main/java/org/iban4j/bban/BbanStructureEntry.java:119]: hi @hajk1 can you help with this TODO please? I'm not sure what to do

**Comment by Bas-Boven** [comment]
We are eagerly expecting this improvement. Would love to see it merged!

File: src/main/java/org/iban4j/bban/BbanStructureEntry.java (hunk 1 above)
Comment (hajk1) [reply src/main/java/org/iban4j/bban/BbanStructureEntry.java:119]: @aSemy sorry I missed this comment. You can keep the "A-Z0-9" which is more logical and also backward compatible.

Comment (aSemy) [reply src/main/java/org/iban4j/bban/BbanStructureEntry.java:119]:
```suggestion
        /**
         * Combined alphabetical (uppercase) and numeric characters (A-Z and 0-9)
         */
        c
```

**Review by hajk1 — APPROVED** [review]

**Pull request merged**